### ISSUE TITLE
Small object detection water surface

### ISSUE THREAD

**Issue by jurvanwijk** [ultralytics/ultralytics#16769]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have a question regarding the optimal training/prediction setup to detect floating litter (single class) on the water surface. 
My dataset consist of images of varying image and object annotations sizes (approx 600 images, 2700 annotations):

Image size groups:
Image size: 4056x3130 -> Average width: 46.46 pixels, Average height: 51.00 pixels
Image size: 2348x2404 -> Average width: 48.27 pixels, Average height: 54.87 pixels
Image size: 2560x1440 -> Average width: 45.03 pixels, Average height: 48.77 pixels
Image size: 1920x1080 -> Average width: 38.82 pixels, Average height: 42.78 pixels
Image size: 1744x1222 -> Average width: 38.79 pixels, Average height: 36.39 pixels
Image size: 2592x1944 -> Average width: 39.99 pixels, Average height: 40.60 pixels
Image size: 1380x1600 -> Average width: 38.50 pixels, Average height: 34.64 pixels
Image size: 2220x2404 -> Average width: 55.93 pixels, Average height: 57.78 pixels
Image size: 2932x2244 -> Average width: 61.23 pixels, Average height: 58.25 pixels
Image size: 2200x2350 -> Average width: 66.62 pixels, Average height: 78.62 pixels
Image size: 2188x2204 -> Average width: 53.90 pixels, Average height: 51.41 pixels
Image size: 4055x1715 -> Average width: 35.67 pixels, Average height: 30.39 pixels
Image size: 3840x2160 -> Average width: 75.29 pixels, Average height: 74.04 pixels
Image size: 1920x1440 -> Average width: 48.82 pixels, Average height: 49.29 pixels
Image size: 4032x3040 -> Average width: 37.51 pixels, Average height: 33.47 pixels

I have tried several approaches:
- yolov8 training with different image sizes (ranging from 640 to 1500)
- tiled training (with separate training of tiles (224x224) and (640x640))
- SSL training with detectron2 (tiled)

All leading to mixed results in quality. There still tends to be a large quantity of false positives regarding the waves in the water, sun glint or shadows, which are being seen as litter by the model.

What would be the ideal training/detection approach to improve this and remove the false positive issues?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jurvanwijk, thank you for reaching out and your interest in Ultralytics 🚀! This is an automated response to guide you while an Ultralytics engineer will also assist soon.

For your small object detection on water surfaces, here are a few suggestions: 

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us investigate.

For your ❓ Question, it sounds like you're dealing with a challenging dataset, considering the false positives. Make sure to explore advanced model tuning techniques and assess the effect of different augmentation strategies. Also, verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For the latest features and enhancements, ensure your environment is up-to-date. You can upgrade to the latest `ultralytics` package with:

```bash
pip install -U ultralytics
```

Explore various verified environments for running YOLO, such as using Notebooks with free GPU, Google Cloud, AWS, or Docker, as detailed in our [environment guides](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).

Join our community for more support and insights! For real-time chat, check out our [Discord](https://ultralytics.com/discord) 🎧. For in-depth discussions, browse our [Discourse](https://community.ultralytics.com). And for community interaction, visit our [Subreddit](https://reddit.com/r/ultralytics).

### Status

Check the current status of our CI tests on the [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) to ensure all is functioning correctly.

We hope this contributes positively to your project. An Ultralytics engineer will follow up with more specific assistance soon! 😊

**Comment by Y-T-G** [comment]
> tiled training (with separate training of tiles (224x224) and (640x640))

If you're using tiled training, you will also need to ensure inference is similarly tiled.

**Comment by jurvanwijk** [comment]
> > tiled training (with separate training of tiles (224x224) and (640x640))
> 
> If you're using tiled training, you will also need to ensure inference is similarly tiled.

Yes, so for the training with tiles I tested tiled images for inference, which also still led to less then great results

**Comment by glenn-jocher** [comment]
Consider experimenting with different augmentation techniques and refining your model's hyperparameters to improve results. If false positives persist, you might also explore using a more specialized model or dataset enhancements.

**Comment by jurvanwijk** [comment]
> Consider experimenting with different augmentation techniques and refining your model's hyperparameters to improve results. If false positives persist, you might also explore using a more specialized model or dataset enhancements.

What hyper parameters and augmentation techniques would you suggest to change/look into?

**Comment by glenn-jocher** [comment]
Focus on adjusting the learning rate, batch size, and image size. For augmentation, try techniques like rotation, brightness adjustment, and contrast enhancement to improve model robustness.

**Comment by jurvanwijk** [comment]
@proevgenii 

Thanks for your reaction. 

What would be the purpose of oversampling with clear water surfaces? The images with "no clear surfaces", (e.g. lots of ripples/foam in the water) are the main cause of FP's. You think the model might improve if empty images of these situations (with no litter) are added to the trainings dataset?

I am using 640 x 640 tiles, created from roughly 4000x3000 original images, using only the tiles containing litter. For detection, I am using SAHI detection, with tile sizes of both 640x640 and 1024x1024 (currently testing different tile sizes for SAHI detection).

**Comment by proevgenii** [comment]
@jurvanwijk 

I meant to add background images without labels(here you know better, I just suggest and idea what can be used as negative/background sample) to reduce FP, model will learn to distinguish between actual objects and typical backgrounds and I'll expect decreasing of FP detections
Does that make more sense now ? 

Also, just to clarify: are you creating tiled dataset using SAHI and then run training with default training setup from Ultralytics?
Or does SAHI provide a custom training loop?
It seems to me there might be some differences, about merging tile detections before passing them to the NMS step during training and it could affect the results

**Comment by jurvanwijk** [comment]
@proevgenii 

Thanks for the reaction. 

So adding background images of situations where a lot of FP's are occurring now? Sounds good, I'll give it a try.

I tile the images and annotations, which I then feed to the yolov8 model for "standard" ultralytics training. For the inference/prediction, I use a custom script, based the SAHI yolov8 script. With the idea that both the training and detection is being done on tiles of 640 or 1024. 

If you have any more suggestions/tips, thanks in advance!
